### PR TITLE
 set_link: fix problem of packets lost during ping on windows guest

### PR DESCRIPTION
**Pull request by kimi1978wy**
Add ip check for windows guest and extend the timeout.

Signed-off-by: Yu Wang <wyu@redhat.com>

id:1510750

### CODE REVIEW

**Comment by kimi1978wy** [comment]
Windows 7 and below guest may need more time to recover ip after reboot, so add a ip check to solve this.

 (1/1) smp_4.2048m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.i386.sp2.io-github-autotest-qemu.set_link.reboot_system_reset.status_off: PASS (1568.13 s)

**Comment by kimi1978wy** [comment]
@xiagao @ybduan pls help to review it again, thanks.
 (1/1) smp_4.2048m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.i386.sp2.io-github-autotest-qemu.set_link.reboot_system_reset.status_off: PASS (1537.05 s)

**Comment by ybduan** [comment]
LGTM.
Acked-by: Yanbin Duan yduan@redhat.com

File: qemu/tests/set_link.py [hunk 1]
```diff
@@ -140,6 +140,32 @@ def guest_interface_operstate_check(expect_status, guest_ifname="",
...
+        is detected or after the specified time is expired.
+
+        :param timeout: total checking time allowed
+        """
+        error_context.context("Start checking guest ip", logging.info)
+        session = vm.wait_for_serial_login()
+        try:
+            end_time = time.time() + timeout
+            mac_addr = vm.get_mac_address()
+            while time.time() < end_time:
```
Comment (hereischen) [inline qemu/tests/set_link.py]: Please try to use `wait_for()` here.

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Pull request merged**